runtime/context.go

```diff
@@ -64,11 +64,12 @@ const (
 	scopeTagHandler         = "handlerid"
 	scopeTagError           = "error"
 	scopeTagStatus          = "status"
-	scopeTagProtocal        = "protocal"
+	scopeTagProtocol        = "protocol"
 	scopeTagHTTP            = "HTTP"
 	scopeTagTChannel        = "TChannel"
 	scopeTagsTargetService  = "targetservice"
 	scopeTagsTargetEndpoint = "targetendpoint"
+	scopeTagResolvedHeaders = "resolvedheaders"
```

Comment:
What is the semantics of this tag? How to decide when it is true or false?

Reply:
This is used for tchannel endpoint.request metrics, currently, if we do `fetch endpoint.request`, it would return http request count + [tchannel request count](https://github.com/uber/zanzibar/blob/1cbe93ed5b0781fc0bb6fe29c684c375a396434b/runtime/tchannel_server.go#L270) + [tchannel request count with resolved headers](https://github.com/uber/zanzibar/blob/1cbe93ed5b0781fc0bb6fe29c684c375a396434b/runtime/tchannel_server.go#L239), this commits corrects the query request count result by query `fetch endpoint.request diff (fetch endpoint.request resolvedheaders:true)`

Reply:
Do we need this counter https://github.com/uber/zanzibar/blob/1cbe93ed5b0781fc0bb6fe29c684c375a396434b/runtime/tchannel_server.go#L270 at all? It seems to be counting the same request as https://github.com/uber/zanzibar/blob/1cbe93ed5b0781fc0bb6fe29c684c375a396434b/runtime/tchannel_server.go#L239, but with less tags.

Reply:
> Do we need this counter https://github.com/uber/zanzibar/blob/1cbe93ed5b0781fc0bb6fe29c684c375a396434b/runtime/tchannel_server.go#L270 at all? It seems to be counting the same request as https://github.com/uber/zanzibar/blob/1cbe93ed5b0781fc0bb6fe29c684c375a396434b/runtime/tchannel_server.go#L239, but with potenitally fewer tags.

If we want to record all endpoint.request(including the errors one) for tchannel, then we need it.

Reply:
The only error we could miss is when it errors here https://github.com/uber/zanzibar/blob/1cbe93ed5b0781fc0bb6fe29c684c375a396434b/runtime/tchannel_server.go#L228. We can emit the request count metrics in the `finish` func https://github.com/uber/zanzibar/blob/1cbe93ed5b0781fc0bb6fe29c684c375a396434b/runtime/tchannel_server.go#L273, because it is always called, no matter there is error or not. I think it is not good to have 2 almost identical counters and a tag whose semantics is hard to reason about without knowing the internal implementation.